forum/models/users.py

```diff
@@ -0,0 +1,133 @@
+"""Users Class for mongo backend."""
+
+from forum.models.abstract import AbstractModel
+
+
+class Users(AbstractModel):
+    """
+    Users class for cs_comments_service user model
+    """
+
+    def __init__(self, collection_name="users", client=None):
+        """
+        Initializes the Users class.
+
+        Args:
+            collection_name: The name of the MongoDB collection.
+            client: The MongoDB client.
+
+        """
+        super().__init__(collection_name, client)
+
+    def get(self, external_id):
+        """
+        Retrieves a user document from the database based on the external_id.
+
+        Args:
+            external_id: The external ID of the user.
+
+        Returns:
+            The user document if found, otherwise None.
+
+        """
+        user_data = self.get_collection.find_one({"external_id": external_id})
+        return user_data
+
+    def insert(
+        self,
+        external_id,
+        username,
+        email,
+        default_sort_key="date",
+        read_states=None,
+        course_stats=None,
+    ):
+        """
+        Inserts a new user document into the database.
+
+        Args:
+            external_id: The external ID of the user.
+            username: The username of the user.
+            email: The email of the user.
+            default_sort_key: The default sort key for the user.
+            read_states: The read states of the user.
+            course_stats: The course statistics of the user.
+
+        Returns:
+            The ID of the inserted document.
+
+        """
+        user_data = {
+            "external_id": external_id,
+            "username": username,
+            "email": email,
+            "default_sort_key": default_sort_key,
+            "read_states": read_states,
+            "course_stats": course_stats,
+        }
+        result = self.get_collection.insert_one(user_data)
+        return str(result.inserted_id)
+
+    def delete(self, external_id):
+        """
+        Deletes a user document from the database based on the external_id.
+
+        Args:
+            external_id: The external ID of the user.
+
+        Returns:
+            The number of documents deleted.
+
+        """
+        result = self.get_collection.delete_one({"external_id": external_id})
+        return result.deleted_count
+
+    def list(self, **kwargs):
+        """
+        Retrieves a list of all user documents in the database.
+
+        Returns:
+            A list of user documents.
+
+        """
+        users_data = self.get_collection.find(**kwargs)
```

Comment:
The find method accepts a dictionary, it should be `kwargs` instead of `**kwargs`.

Reply:
yes typo, fixed